packages/salesforce/src/Adaptor.js

```diff
@@ -317,7 +317,9 @@ export function bulkQuery(qs, options, callback) {
       qs,
       options
     );
-    const apiVersion = connection.version;
+
+    if (connection.version < 47.0)
```

Comment:
`connection.version` returns a string, so this test likely won't work if you pass in `apiVersion: "46"`

Reply:
Thanks @josephjclark, i have updated this to `connection.version < '47.0'`.

In other news i have learnt more about Javascript implicit type conversation 😀 ,
![Screenshot 2024-05-23 at 1 31 24 PM](https://github.com/OpenFn/adaptors/assets/6592749/0a7bce71-edb8-4979-bb79-e94bc0e84687)